scylla/src/cloud/mod.rs

```diff
@@ -1,4 +1,59 @@
 mod config;
 
+use std::net::IpAddr;
+
 pub(crate) use config::CloudConfig;
 pub use config::CloudConfigError;
+use openssl::{
+    error::ErrorStack,
+    ssl::{SslContext, SslMethod, SslVerifyMode},
+};
+use tracing::warn;
+use uuid::Uuid;
+
+use crate::transport::connection::{ConnectionConfig, SslConfig};
+
+pub(crate) fn set_sni_ssl_config_for_scylla_cloud(
+    host_id: Option<Uuid>,
+    dc: Option<&str>,
+    address: IpAddr,
+    port: u16,
+    connection_config: &mut ConnectionConfig,
```

Comment:
Describe SslConfig field mutations (so that all 3 possible flows are documented - no SSL, user-provided SslContext and serverless SSL configuration).